### PR TITLE
refactor `data_type/can_cast`

### LINKED ISSUE #1688: can_cast

### PR DESCRIPTION
**Pull request by flych3r**
Added examples to `ivy/functional/ivy/data_type/can_cast`

Closes #1688

### CODE REVIEW

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

The examples look good, just a few additional things you have to do:
1. Instance methods for `ivy.array` and `ivy.container` needs to be added. The instance method for `ivy.array` will be added in `ivy/array/data_types.py` and for container it will be added in `ivy/container/data_types.py`. You can use [this](https://github.com/unifyai/ivy/blob/76d65119fef75a3592a6e2cd744c568ea73f7651/ivy/array/elementwise.py#L295) example of tan as a reference to complete can_cast.
2. You have to add docstring examples for `ivy.array` instance method and `ivy.container` method, These examples will be placed under the new instance function you will create in their respective folders(`ivy/array/data_types.py` for array and `ivy/container/data_types.py`).
3. And lastly you also have to add a static_method for can_cast in `ivy/container/data_types.py`. [here](https://github.com/unifyai/ivy/blob/76d65119fef75a3592a6e2cd744c568ea73f7651/ivy/container/elementwise.py#L1372) here is an example of static_tan. You also have to write 1 example of the new static function.

All of these requirements are mentioned [here](https://github.com/unifyai/ivy/blob/76d65119fef75a3592a6e2cd744c568ea73f7651/docs/partial_source/deep_dive/11_docstring_examples.rst) in details.
If you have any question then feel free to ask, I'd be happy to help 😄 .

File: ivy/array/data_types.py [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+        """
+        `ivy.Array` instance method variant of `ivy.can_cast`. This method simply wraps
+        the function, and so the docstring for `ivy.can_cast` also applies to this
+        method with minimal changes.
+
+        Examples
+        --------
+        >>> x = ivy.array([1., 2., 3.])
+        >>> print(x.dtype)
+        float32
```
Comment (sherry30) [inline ivy/array/data_types.py:21]: add new line after this

File: ivy/container/data_types.py [hunk 2]
```diff
@@ -1,9 +1,74 @@
...
+        """
+        `ivy.Container` static method variant of `ivy.can_cast`. This method simply
+        wraps the function, and so the docstring for `ivy.can_cast` also applies to
+        this method with minimal changes.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3, 4, 5]))
+        >>> print(x.a.dtype, x.b.dtype)
+        float32 int32
```
Comment (sherry30) [inline ivy/container/data_types.py:31]: add new line after this

File: ivy/container/data_types.py [hunk 3]
```diff
@@ -1,9 +1,74 @@
...
+        """
+        `ivy.Container` instance method variant of `ivy.can_cast`. This method simply
+        wraps the function, and so the docstring for `ivy.can_cast` also applies to
+        this method with minimal changes.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3, 4, 5]))
+        >>> print(x.a.dtype, x.b.dtype)
+        float32 int32
```
Comment (sherry30) [inline ivy/container/data_types.py:65]: add new line after this

File: ivy/functional/ivy/data_type.py [hunk 4]
```diff
@@ -46,6 +46,49 @@ def can_cast(
...
+    True
+
+    >>> print(ivy.can_cast(ivy.float64, 'int64'))
+    False
+
+    With :code:`ivy.Array` input:
+
+    >>> x = ivy.array([1., 2., 3.])
+    >>> print(x.dtype)
+    float32
```
Comment (sherry30) [inline ivy/functional/ivy/data_type.py:72]: add new line after this

File: ivy/functional/ivy/data_type.py [hunk 5]
```diff
@@ -46,6 +46,49 @@ def can_cast(
...
+    >>> print(ivy.can_cast(x, ivy.float64))
+    True
+
+    >>> x = ivy.native_array([[-1, -1, -1], [1, 1, 1]], dtype='int16')
+    >>> print(x.dtype)
+    int16
```
Comment (sherry30) [inline ivy/functional/ivy/data_type.py:78]: write this as `<dtype:'int16'>`

File: ivy/functional/ivy/data_type.py [hunk 6]
```diff
@@ -46,6 +46,49 @@ def can_cast(
...
+
+    With :code:`ivy.Array` input:
+
+    >>> x = ivy.array([1., 2., 3.])
+    >>> print(x.dtype)
+    float32
+    >>> print(ivy.can_cast(x, ivy.float64))
+    True
+
+    >>> x = ivy.native_array([[-1, -1, -1], [1, 1, 1]], dtype='int16')
```
Comment (sherry30) [inline ivy/functional/ivy/data_type.py:76]: 1. add `With :code:'ivy.NativeArray' input:` before this
2. write dtype input on a new line using `\` separator

File: ivy/functional/ivy/data_type.py [hunk 7]
```diff
@@ -46,6 +46,49 @@ def can_cast(
...
+    >>> print(x.dtype)
+    int16
+    >>> print(ivy.can_cast(x, 'uint8'))
+    False
+
+    With :code:`ivy.Container` input:
+
+    >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3, 4, 5]))
+    >>> print(x.a.dtype, x.b.dtype)
+    float32 int32
```
Comment (sherry30) [inline ivy/functional/ivy/data_type.py:86]: add new line after this

File: ivy/functional/ivy/data_type.py [hunk 8]
```diff
@@ -46,6 +46,49 @@ def can_cast(
...
+
+    >>> x = ivy.native_array([[-1, -1, -1], [1, 1, 1]], dtype='int16')
+    >>> print(x.dtype)
+    int16
+    >>> print(ivy.can_cast(x, 'uint8'))
+    False
+
+    With :code:`ivy.Container` input:
+
+    >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3, 4, 5]))
```
Comment (sherry30) [inline ivy/functional/ivy/data_type.py:84]: write b input on a next line using `\` separator

File: ivy/container/data_types.py [hunk 9]
```diff
@@ -1,9 +1,74 @@
...
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+        """
+        `ivy.Container` instance method variant of `ivy.can_cast`. This method simply
+        wraps the function, and so the docstring for `ivy.can_cast` also applies to
+        this method with minimal changes.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3, 4, 5]))
```
Comment (sherry30) [inline ivy/container/data_types.py:63]: write b input on the next line using `\` separator

File: ivy/container/data_types.py [hunk 10]
```diff
@@ -1,9 +1,74 @@
...
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+        """
+        `ivy.Container` static method variant of `ivy.can_cast`. This method simply
+        wraps the function, and so the docstring for `ivy.can_cast` also applies to
+        this method with minimal changes.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([0., 1., 2.]), b=ivy.array([3, 4, 5]))
```
Comment (sherry30) [inline ivy/container/data_types.py:29]: write b input on the next line using `\` separator

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

can_cast does not accept `out` argument so you're right about not adding that. Other than that
your docstring seems to be failing, its because you haven't added a new line after writing the output so just a few minor changes you need to fix that I've mentioned on each specific line.
I'll merge your PR as soon as you fix these 😃 

Thanks

**Comment by flych3r** [comment]
Hello, just made the requested changes. One thing, the item of the function `can_cast` on issue #780 was not marked when I made the comment.

**Pull request merged**